sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4320,33 +4289,48 @@ def _check_existing_cluster(
             # TODO(zhwu): complete the list of exceptions.
         """
         record = global_user_state.get_cluster_from_name(cluster_name)
-        handle_before_refresh = None if record is None else record['handle']
-        status_before_refresh = None if record is None else record['status']
+        if record is None:
+            handle_before_refresh = None
+            status_before_refresh = None
+        else:
+            handle_before_refresh = record['handle']
+            status_before_refresh = record['status']
 
         prev_cluster_status, handle = (status_before_refresh,
                                        handle_before_refresh)
 
         if not dryrun:
-            prev_cluster_status, handle = (
-                backend_utils.refresh_cluster_status_handle(
-                    cluster_name,
-                    # We force refresh for the init status to determine the
-                    # actual state of a previous cluster in INIT state.
-                    #
-                    # This is important for the case, where an existing cluster
-                    # is transitioned into INIT state due to key interruption
-                    # during launching, with the following steps:
-                    # (1) launch, after answering prompt immediately ctrl-c;
-                    # (2) launch again.
-                    # If we don't refresh the state of the cluster and reset it
-                    # back to STOPPED, our failover logic will consider it as an
-                    # abnormal cluster after hitting resources capacity limit on
-                    # the cloud, and will start failover. This is not desired,
-                    # because the user may want to keep the data on the disk of
-                    # that cluster.
-                    force_refresh_statuses={status_lib.ClusterStatus.INIT},
-                    acquire_per_cluster_status_lock=False,
-                ))
+            record = backend_utils.refresh_cluster_record(
+                cluster_name,
+                # We force refresh for the init status to determine the
+                # actual state of a previous cluster in INIT state.
+                #
+                # This is important for the case, where an existing cluster
+                # is transitioned into INIT state due to key interruption
+                # during launching, with the following steps:
+                # (1) launch, after answering prompt immediately ctrl-c;
+                # (2) launch again.
+                # If we don't refresh the state of the cluster and reset it
+                # back to STOPPED, our failover logic will consider it as an
+                # abnormal cluster after hitting resources capacity limit on
+                # the cloud, and will start failover. This is not desired,
+                # because the user may want to keep the data on the disk of
+                # that cluster.
```

Comment:
Since now we have the "ever up" logic, is this comment still accurate? Do we still need to force-refresh for INIT (I guess it doesn't harm to properly restore it to STOPPED for nicer UX, but perhaps the avoid-failover reason no longer applies)?